service/service.go

```diff
@@ -73,19 +74,44 @@ func (app *Application) ReportFatalError(err error) {
 	app.asyncErrorChannel <- err
 }
 
-// New creates and returns a new instance of Application
+// New creates and returns a new instance of Application.
 func New(
 	factories config.Factories,
 ) *Application {
-	return &Application{
+	app := &Application{
 		v:         viper.New(),
 		readyChan: make(chan struct{}),
 		factories: factories,
 	}
+
+	rootCmd := &cobra.Command{
+		Use:  "otelcol",
+		Long: "OpenTelemetry Collector",
+		Run: func(cmd *cobra.Command, args []string) {
+			app.init()
+			app.execute()
+		},
+	}
+
+	// TODO: coalesce this code and expose this information to other components.
```

Comment:
As explained on the description let's discuss and do a better treatment for the command-line parameters in a separate PR.